_drafts/practical_zephyr_01_kconfig.md

```diff
@@ -0,0 +1,986 @@
+                 name                 | current | recommended || check result
+==============================================================================
+CONFIG_OVERRIDE_FRAME_POINTER_DEFAULT |    n    |      y      ||     FAIL
+CONFIG_USE_SEGGER_RTT                 |    y    |      n      ||     FAIL
+CONFIG_BUILD_OUTPUT_STRIPPED          |    n    |      y      ||     FAIL
+CONFIG_FAULT_DUMP                     |    2    |      0      ||     FAIL
+CONFIG_STACK_SENTINEL                 |    n    |      y      ||     FAIL
+```
+
+The symbols in `prj_release.conf` have been chosen such that at the time of writing all hardening options are fulfilled and the above table is empty.
+
+
+
+## A custom Kconfig symbol
+
+In the previous sections, we had a thorough look at _Kconfig_ and its workings. To wrap up this article, we'll create a custom, application-specific symbol and use it in our application _and_ build process. While this is a rather unlikely use case, it nicely demonstrates how _Kconfig_ works.
+
+> **Note:** This section borrows from the [nRF Connect SDK Fundamentals lesson on configuration files](https://academy.nordicsemi.com/courses/nrf-connect-sdk-fundamentals/lessons/lesson-3-elements-of-an-nrf-connect-sdk-application/topic/configuration/) that is freely available in the [Nordic Developer Academy](https://academy.nordicsemi.com/). If you're looking for additional challenges, check out the available courses!
+
+
+### Adding a custom configuration
+
+[CMake automatically detects a `Kconfig` file](https://docs.zephyrproject.org/latest/develop/application/index.html#application-cmakelists-txt) if it is placed in the same directory of the application's `CMakeLists.txt`, and that is what we'll use for our own configuration file. In case you want to place the `Kconfig` file somewhere else, you can customize this behavior using an absolute path for the `KCONFIG_ROOT` build system variable.
+
+Similar to the file `zephyr/subsys/debug/Kconfig` of Zephyr's _debug_ subsystem, the `Kconfig` file specifies all available configuration options and their dependencies. We'll only specify a simple boolean symbol in the `Kconfig` file in our application's root directory. Please refer to the [official documentation](https://docs.zephyrproject.org/latest/build/kconfig/setting.html#setting-kconfig-configuration-values) for details about the `Kconfig` syntax:
+
+
+```bash
+$ tree --charset=utf-8 --dirsfirst -L 1
+.
+├── boards
+├── src
+├── CMakeLists.txt
+├── Kconfig
+├── extra0.conf
+├── extra1.conf
+├── prj.conf
+└── prj_release.conf
+
+$ cat Kconfig
+```
+```conf
+mainmenu "Customized Menu Name"
+source "Kconfig.zephyr"
+
+menu "Application Options"
+config USR_FUN
+	bool "Enable usr_fun"
+	default n
+	help
+	  Enables the usr_fun function.
+endmenu
+```
+
+The skeleton of this `Kconfig` file is well explained in the [official documentation](https://docs.zephyrproject.org/latest/develop/application/index.html#application-cmakelists-txt):
```

Comment:
Sad there's not a better anchor to add to the link here, I might mention to "see step 4 in the Application CMakeLists.txt section"

Reply:
Yep. I'm also a bit afraid that all the links will be useless in the future ... Do you have a little helper script to extract and just run a quick `GET` on references in the interrupt blog to make sure they're not dead links?